### ISSUE TITLE
this.createFix is not a function (5.4.0)

### ISSUE THREAD

**Issue by jlowcs** [buzinas/tslint-eslint-rules#186]
I'm getting this error with version 3.4.0:

```
TypeError: this.createFix is not a function
  at ObjectCurlySpacingWalker.checkSpacingInsideBraces (E:\dev\projects\postfiles\pf-frontend\node_modules\tslint-eslint-rules\dist\rules\objectCurlySpacingRule.js:61:32)
  at ObjectCurlySpacingWalker.visitNode (E:\dev\projects\postfiles\pf-frontend\node_modules\tslint-eslint-rules\dist\rules\objectCurlySpacingRule.js:47:18)
  at E:\dev\projects\postfiles\pf-frontend\node_modules\tslint\lib\language\walker\syntaxWalker.js:517:63
  at visitEachNode (E:\dev\projects\postfiles\pf-frontend\node_modules\typescript\lib\typescript.js:14209:30)
  at Object.forEachChild (E:\dev\projects\postfiles\pf-frontend\node_modules\typescript\lib\typescript.js:14335:21)
  at ObjectCurlySpacingWalker.SyntaxWalker.walkChildren (E:\dev\projects\postfiles\pf-frontend\node_modules\tslint\lib\language\walker\syntaxWalker.js:517:12)
  at ObjectCurlySpacingWalker.SyntaxWalker.visitCallExpression (E:\dev\projects\postfiles\pf-frontend\node_modules\tslint\lib\language\walker\syntaxWalker.js:53:14)
  at ObjectCurlySpacingWalker.SyntaxWalker.visitNode (E:\dev\projects\postfiles\pf-frontend\node_modules\tslint\lib\language\walker\syntaxWalker.js:295:22)
  at ObjectCurlySpacingWalker.visitNode (E:\dev\projects\postfiles\pf-frontend\node_modules\tslint-eslint-rules\dist\rules\objectCurlySpacingRule.js:49:36)
  at E:\dev\projects\postfiles\pf-frontend\node_modules\tslint\lib\language\walker\syntaxWalker.js:517:63
  at visitNode (E:\dev\projects\postfiles\pf-frontend\node_modules\typescript\lib\typescript.js:14197:20)
  at Object.forEachChild (E:\dev\projects\postfiles\pf-frontend\node_modules\typescript\lib\typescript.js:14364:24)
  at ObjectCurlySpacingWalker.SyntaxWalker.walkChildren (E:\dev\projects\postfiles\pf-frontend\node_modules\tslint\lib\language\walker\syntaxWalker.js:517:12)
  at ObjectCurlySpacingWalker.SyntaxWalker.visitNode (E:\dev\projects\postfiles\pf-frontend\node_modules\tslint\lib\language\walker\syntaxWalker.js:511:22)
  at ObjectCurlySpacingWalker.visitNode (E:\dev\projects\postfiles\pf-frontend\node_modules\tslint-eslint-rules\dist\rules\objectCurlySpacingRule.js:49:36)
  at E:\dev\projects\postfiles\pf-frontend\node_modules\tslint\lib\language\walker\syntaxWalker.js:517:63
  at visitEachNode (E:\dev\projects\postfiles\pf-frontend\node_modules\typescript\lib\typescript.js:14209:30)
  at Object.forEachChild (E:\dev\projects\postfiles\pf-frontend\node_modules\typescript\lib\typescript.js:14335:21)
  at ObjectCurlySpacingWalker.SyntaxWalker.walkChildren (E:\dev\projects\postfiles\pf-frontend\node_modules\tslint\lib\language\walker\syntaxWalker.js:517:12)
  at ObjectCurlySpacingWalker.SyntaxWalker.visitCallExpression (E:\dev\projects\postfiles\pf-frontend\node_modules\tslint\lib\language\walker\syntaxWalker.js:53:14)
  at ObjectCurlySpacingWalker.SyntaxWalker.visitNode (E:\dev\projects\postfiles\pf-frontend\node_modules\tslint\lib\language\walker\syntaxWalker.js:295:22)
  at ObjectCurlySpacingWalker.visitNode (E:\dev\projects\postfiles\pf-frontend\node_modules\tslint-eslint-rules\dist\rules\objectCurlySpacingRule.js:49:36)
  at E:\dev\projects\postfiles\pf-frontend\node_modules\tslint\lib\language\walker\syntaxWalker.js:517:63
  at visitNode (E:\dev\projects\postfiles\pf-frontend\node_modules\typescript\lib\typescript.js:14197:20)
  at Object.forEachChild (E:\dev\projects\postfiles\pf-frontend\node_modules\typescript\lib\typescript.js:14391:24)
  at ObjectCurlySpacingWalker.SyntaxWalker.walkChildren (E:\dev\projects\postfiles\pf-frontend\node_modules\tslint\lib\language\walker\syntaxWalker.js:517:12)
  at ObjectCurlySpacingWalker.SyntaxWalker.visitExpressionStatement (E:\dev\projects\postfiles\pf-frontend\node_modules\tslint\lib\language\walker\syntaxWalker.js:107:14)
  at ObjectCurlySpacingWalker.SyntaxWalker.visitNode (E:\dev\projects\postfiles\pf-frontend\node_modules\tslint\lib\language\walker\syntaxWalker.js:349:22)
  at ObjectCurlySpacingWalker.visitNode (E:\dev\projects\postfiles\pf-frontend\node_modules\tslint-eslint-rules\dist\rules\objectCurlySpacingRule.js:49:36)
  at E:\dev\projects\postfiles\pf-frontend\node_modules\tslint\lib\language\walker\syntaxWalker.js:517:63
  at visitEachNode (E:\dev\projects\postfiles\pf-frontend\node_modules\typescript\lib\typescript.js:14209:30)
  at Object.forEachChild (E:\dev\projects\postfiles\pf-frontend\node_modules\typescript\lib\typescript.js:14382:24)
  at ObjectCurlySpacingWalker.SyntaxWalker.walkChildren (E:\dev\projects\postfiles\pf-frontend\node_modules\tslint\lib\language\walker\syntaxWalker.js:517:12)
  at ObjectCurlySpacingWalker.SyntaxWalker.visitSourceFile (E:\dev\projects\postfiles\pf-frontend\node_modules\tslint\lib\language\walker\syntaxWalker.js:218:14)
  at ObjectCurlySpacingWalker.SyntaxWalker.visitNode (E:\dev\projects\postfiles\pf-frontend\node_modules\tslint\lib\language\walker\syntaxWalker.js:463:22)
  at ObjectCurlySpacingWalker.visitNode (E:\dev\projects\postfiles\pf-frontend\node_modules\tslint-eslint-rules\dist\rules\objectCurlySpacingRule.js:49:36)
  at ObjectCurlySpacingWalker.SyntaxWalker.walk (E:\dev\projects\postfiles\pf-frontend\node_modules\tslint\lib\language\walker\syntaxWalker.js:23:14)
  at Rule.AbstractRule.applyWithWalker (E:\dev\projects\postfiles\pf-frontend\node_modules\tslint\lib\language\rule\abstractRule.js:44:16)
  at Rule.apply (E:\dev\projects\postfiles\pf-frontend\node_modules\tslint-eslint-rules\dist\rules\objectCurlySpacingRule.js:17:21)
  at Linter.applyRule (E:\dev\projects\front\tools\groot\node_modules\tslint\lib\linter.js:143:33)
  at Linter.lint (E:\dev\projects\front\tools\groot\node_modules\tslint\lib\linter.js:109:41)
  at files.map (E:\dev\projects\front\tools\groot\lib\modules\lint\tslint.js:44:11)
  at Array.map (native)
  at Object.module.exports [as checkTsLint] (E:\dev\projects\front\tools\groot\lib\modules\lint\tslint.js:40:9)
  at Promise.resolve.then.then.then (E:\dev\projects\front\tools\groot\lib\modules\lint\index.js:43:30)
  at process._tickCallback (internal/process/next_tick.js:103:7)
  at Module.runMain (module.js:606:11)
  at run (bootstrap_node.js:394:7)
  at startup (bootstrap_node.js:149:9)
  at bootstrap_node.js:509:3
```

Seems like it's linked to missing spaces in curly braces:

```
angular.module('pf.ng2', []).directive(
	'pfNg2Test',
    downgradeComponent({component: TestComponent}) as angular.IDirectiveFactory,
);
```

The following fixed the exception:
```
angular.module('pf.ng2', []).directive(
	'pfNg2Test',
    downgradeComponent({ component: TestComponent }) as angular.IDirectiveFactory,
);
```

**Comment by jlowcs** [comment]
Here are my rules:
```javascript
	"rules": {
		"no-inner-declarations": [true, "both"],
		"no-multi-spaces": [true],
		"array-bracket-spacing": [true, "never"],
		"block-spacing": [true, "always"],
		"brace-style": [true, "1tbs", { "allowSingleLine": true }],
		"ter-indent": [true, "tab", { "MemberExpression": 1, "SwitchCase": 1 }],
		"object-curly-spacing": [true, "always"],
		"ter-arrow-spacing": [true, { "before": true, "after": true }],
		"no-extra-boolean-cast": true,
		"no-constant-condition": true,
		"no-control-regex": true,
		"no-duplicate-case": true,
		"no-empty-character-class": true,
		"no-ex-assign": true,
		"no-extra-semi": true,
		"no-invalid-regexp": true,
		"no-irregular-whitespace": true,
		"no-regex-spaces": true,
		"no-sparse-arrays": true,
		"no-unexpected-multiline": true,
		"valid-typeof": true
	}
```

**Comment by jmlopez-rod** [comment]
What tslint version are you using?

**Comment by jlowcs** [comment]
tslint 4.2.0

Just tested with a newer version, and it works.

**Comment by jmlopez-rod** [comment]
Guess it's time to update the peer dependency so that it warns us of this issue.

**Comment by jmlopez-rod** [comment]
Turns out the `createFix` method was introduced here: https://github.com/palantir/tslint/commit/e933f7d85b120041b64c5ce7edcfbbdf80e57926 which is only available starting tslint 4.3.0.

**Comment by jmlopez-rod** [comment]
On the next release npm should warn users that `tslint-eslint-rules` requires tslint to be at least 4.3.0.
